### DISCUSSION TITLE
Parsing STUFF with SELECT inside throws NULL exception

### DISCUSSION THREAD

**Discussion by cptb** [bruce-dunwiddie/tsql-parser#107] [General]
Hi Bruce,
I am sorry for "always" throwing hard questions at you. This tool is wonderful and I have been using it since last year.
Just now, I met a complex query which contains a STUFF function, and inside the function, the first parameter is a SELECT statement to generate some texts. It looks like the below statement.
`select stuff
(
    (select top 10 note from asset_note where note is not null order by note_id for XML path(''))
    , 1
    , 1
    , 'Note: '
);`  

My C# code looks like this:

```
string sql = @"select stuff((select top 10 note from asset_note where note is not null order by note_id for xml path('')), 1, 1 , 'Note: '); ";

var statements = TSQLStatementReader.ParseStatements(sql); // this method throws NULL ref exception
```

If the script is simple like this:
`SELECT STUFF('SQL Tutorial', 1, 3, 'HTML')`
It can be parsed successfully.

Could you please look at this issue to see if it is possible to embed SELECT inside the STUFF function?
It is valid SQL to use SELECT inside STUFF.

Regards
Paul

**Comment by bruce-dunwiddie** [comment]
Yep, I'll look into it. I can't offhand remember how deeply implemented "for xml path" is, and I don't remember "path" having options, so that may be the issue there.

I would expect the rest to work as is.

**Comment by bruce-dunwiddie** [comment]
I've confirmed the issue and narrowed the problem down to a very simplistic current implementation inside the FOR clause parser.

I should be able to resolve this within a few days.

**Comment by bruce-dunwiddie** [comment]
Please verify that it's fixed in https://www.nuget.org/packages/TSQL.Parser/2.5.0.

**Reply by cptb** [reply]
Thank you Bruce! With the latest version, the issue has been resolved.
